src/Sfa.Roatp.Indexer.ApplicationServices/ProviderIndexer.cs

```diff
@@ -0,0 +1,149 @@
+            }
+
+            foreach (var roatpProvider in roatpProviders)
+            {
+                var oldRoatpProvider = oldProviders.FirstOrDefault(x => x.Ukprn == roatpProvider.Ukprn);
+
+                if (oldRoatpProvider == null)
+                {
+                    return true;
+                }
+
+                if (!roatpProvider.IsEqual(oldRoatpProvider))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public bool CreateIndex(string indexName)
+        {
+            var indexExists = _indexMaintainer.IndexExists(indexName);
+
+            // If it already exists and is empty, let's delete it.
+            if (indexExists)
+            {
+                _log.Warn("Index already exists, deleting and creating a new one");
+
+                _indexMaintainer.DeleteIndex(indexName);
+            }
+
+            _indexMaintainer.CreateIndex(indexName);
+
+            return _indexMaintainer.IndexExists(indexName);
+        }
+
+        public bool IsIndexCorrectlyCreated(string indexName)
+        {
+            return _indexMaintainer.IndexContainsDocuments(indexName);
+        }
+
+        public void ChangeUnderlyingIndexForAlias(string newIndexName)
+        {
+            if (!_indexMaintainer.AliasExists(_settings.IndexesAlias))
+            {
+                _log.Warn("Alias doesn't exist, creating a new one...");
+
+                _indexMaintainer.CreateIndexAlias(_settings.IndexesAlias, newIndexName);
+            }
+            else
+            {
+                _indexMaintainer.SwapAliasIndex(_settings.IndexesAlias, newIndexName);
+            }
+        }
+
+        public List<RoatpProvider> LoadEntries()
+        {
+            _log.Debug("Loading data at RoATP provider index");
+
+            return _providerDataService.GetRoatpData();
+        }
+
+        public async Task IndexEntries(string indexName, List<RoatpProvider> roatpProviders)
+        {
+            var bulkApiProviderTasks = new List<Task<IBulkResponse>>();
+
+            _log.Debug($"Received {roatpProviders.Count} RoATP providers");
+
+            _log.Debug("Indexing " + roatpProviders.Count + " RoATP providers");
+            bulkApiProviderTasks.AddRange(_indexMaintainer.IndexRoatpProviders(indexName, roatpProviders));
+
+            _indexMaintainer.LogResponse(await Task.WhenAll(bulkApiProviderTasks), "RoatpProviderDocument");
+        }
+
+        public List<RoatpProviderDocument> CheckNewProviders(string newIndexName)
+        {
+            var actualRoatpProviders = _indexMaintainer.LoadRoatpProvidersFromIndex(newIndexName);
+            var oldProviders = _indexMaintainer.LoadRoatpProvidersFromAlias();
+
+            return (from roatpProviderDocument in actualRoatpProviders let roatpProvider = oldProviders.FirstOrDefault(x => x.Ukprn == roatpProviderDocument.Ukprn) where roatpProvider == null select roatpProviderDocument).ToList();
```

Comment:
Could you break this line across three lines have a line break befeore 'let' and 'where'

Also you do not need ToList again here, we can return IEnumerable which gets passed to the SendNewProviderEvent method